### PR TITLE
Add definitions for `json-schema-generator`

### PR DESCRIPTION
**Pull request by marcelltoth**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/json-schema-generator/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *~ This file should be imported using the CommonJS-style:
+ *~   import x = require('someLibrary');
+ *~
+ *~ Refer to the documentation to understand common
+ *~ workarounds for this limitation of ES6 modules.
+ */
+
+/*~ This declaration specifies that the function
+ *~ is the exported object from the file
+ */
```
Comment (peterblazejewicz) [inline types/json-schema-generator/index.d.ts:18]:
```suggestion
```
please remove this, it's just template instruction, could perplex users.

File: types/json-schema-generator/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+export = generate;
```
Comment (peterblazejewicz) [inline types/json-schema-generator/index.d.ts:20]:
```suggestion
export = jsonToSchema;
```
if you switch to that one, everyone would understand that you are following naming convention from teh source and follow source module details:
https://github.com/krg7880/json-schema-generator/blob/27d5a84721c4cd496a2a37bbefda09c07a59f6a1/lib/index.js#L7
it's a nice touch

File: types/json-schema-generator/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+/*~ This example shows how to have multiple overloads for your function */
```
Comment (peterblazejewicz) [inline types/json-schema-generator/index.d.ts:22]:
```suggestion
/* JSON schema generated based on draft-v4 of the specification. */
```

File: types/json-schema-generator/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [inline types/json-schema-generator/index.d.ts:20]:
```suggestion
export = jsonToSchema;
```
ditton about naming, apply only if you agree

File: types/json-schema-generator/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+declare function generate(schema: object): JSONSchema4;
```
Comment (peterblazejewicz) [inline types/json-schema-generator/index.d.ts:23]:
```suggestion
declare function jsonToSchema(json: object): JSONSchema4;
```
ditton about naming, apply only if you agree.

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
LGTM!
@marcelltoth can you just review my comments and apply, where you agree?

File: types/json-schema-generator/index.d.ts (hunk 2 above)
Comment (marcelltoth) [reply types/json-schema-generator/index.d.ts:20]: Makes sense, I'll update it.

**Comment by marcelltoth** [comment]
Thank you for the feedback @peterblazejewicz ! They were great observations, I agree with them all.

Let me know if I missed something.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@marcelltoth thx for the contribution!

**Comment by marcelltoth** [comment]
Ready to merge

**Comment by orta** [comment]
Looks good to me!

**Pull request merged**